app/models/user.rb

```diff
@@ -1,6 +1,23 @@
 class User < ActiveRecord::Base
 
-  def fullname
-    first_name + " " + last_name
+  has_many :massages,    inverse_of: :user
+  has_many :treatments,  inverse_of: :user
+  has_one  :facial_hair, inverse_of: :user
+
+  validates :first_name, :last_name, :email, presence: true
+  validates_uniqueness_of :email
+
+  def self.search(needle)
```

Comment:
```needle``` seems like an odd name.  I'm assuming "needle in a haystack"?  That is a bit clever.  How about ```search_term``` or ```query``` instead?